docs/source/docker/integration/dagster_aws.rst

```diff
@@ -0,0 +1,493 @@
+        
+        """
+
+        def __init__(self):
+            self.api = PyPexels(api_key=PEXELS_API_KEY)
+
+
+        def random_filename(self, size_: int = 8):
+            """Generates a random filename of uppercase letters and digits.
+            
+            """
+            chars = string.ascii_uppercase + string.digits
+            return ''.join(random.choice(chars) for _ in range(size_)) + '.mp4'
+
+
+        def download_video(self, root: str):
+            """Downloads a random popular video from pexels and saves it.
+            
+            """
+            popular_videos = self.api.videos_popular(per_page=40)._body['videos']
+            video = random.choice(popular_videos)
+            video_file = video['video_files'][0]
+            video_link = video_file['link']
+            
+            video = requests.get(video_link)
+            
+            path = os.path.join(root, self.random_filename())
+            with open(path, 'wb') as outfile:
+                outfile.write(video._content)
+
+            return path
+
+
+    @solid
+    def download_random_video_from_pexels() -> str:
+        """Dagster solid to download a random pexels video to the current directory.
+
+        Returns:
+            The path to the downloaded video.
+
+        """
+
+        client = PexelsClient()
+        path = client.download_video('./')
+
+        return path
+
+
+The next solid in the pipeline uploads the video to the S3 bucket. It saves the video
+in a randomly created subfolder in the S3 bucket and passes the object name to the next solid.
+Set the `BUCKET_NAME` and `REGION_NAME` to your bucket name and region of the EC2 instance. 
+
+
+.. code-block:: python
+
+    import os
+    import string
+    import random
+
+    import boto3
+    from botocore.exceptions import ClientError
+
+    from dagster import solid
+
+
+    BUCKET_NAME: str = 'lightly-aws-bucket'
+    REGION_NAME: str = 'YOUR_REGION_NAME' # e.g. eu-central-1
+
+
+    class S3Client:
+        """S3 client to upload files to a bucket.
+        
+        """
+
+        def __init__(self):
+            self.s3 = boto3.client('s3', region_name=REGION_NAME)
+
+
+        def random_subfolder(self, size_: int = 8):
+            """Generates a random subfolder name of uppercase letters and digits.
+            
+            """
+            chars = string.ascii_uppercase + string.digits
+            return ''.join(random.choice(chars) for _ in range(size_))
+
+
+        def upload_file(self, filename: str):
+            """Uploads the file at filename to the s3 bucket.
+
+            Generates a random subfolder so the file will be stored at:
+            >>> BUCKET_NAME/RANDOM_SUBFOLDER/basefilename.mp4
+            
+            """
+
+            # upload file to lightly-aws-bucket/input_dir/RANDOM_STRING/basename.mp4
+            object_name = os.path.join(
+                'input_dir',
+                self.random_subfolder(),
```

Comment:
Why should it generate a random subfolder? Why not call the subfolder always `lightly_data_input` or similar?

Reply:
Because we require one input folder per video for this workflow.